jwql/utils/engineering_database.py

```diff
@@ -0,0 +1,174 @@
+#! /usr/bin/env python
+"""Module to interface the JWST Engineering Database.
+
+This module provides ``jwql`` with functions to interface and query the
+ JWST DMS Engineering Database.
+
+Authors
+-------
+
+    - Johannes Sahlmann
+
+Use
+---
+
+    This module can be imported and used with
+
+    ::
+
+        from jwql.engineering_database import query_single_mnemonic
+        query_single_mnemonic(mnemonic_identifier, start_time, end_time)
+
+    Required arguments:
+
+    ``mnemonic_identifier`` - String representation of a mnemonic name.
+    ``start_time`` - astropy.time.Time instance
+    ``end_time`` - astropy.time.Time instance
+
+Notes
+-----
+    This module is built on top of astroquery.mast and uses JWST-specific MAST services.
+    The user has to provide a valid MAST authentication token.
+
+"""
+import astropy
+from astropy.table import Table
+from astroquery.mast import Mast
+
+from .utils import get_config
+
+# should use oauth.register_oauth()?
+settings = get_config()
+mast_token = settings['mast_token']
+
+Mast.login(token=mast_token)
+
+# could eventually be moved to constants.py
+mast_edb_timeseries_service = 'Mast.JwstEdb.GetTimeseries.All'
+mast_edb_dictionary_service = 'Mast.JwstEdb.Dictionary'
+mast_edb_mnemonic_service = 'Mast.JwstEdb.Mnemonics'
+
+
+class EdbMnemonic():
+    """Class to hold and manipulate results of EngDB queries."""
+
+    def __init__(self, mnemonic_identifier, start_time, end_time, data, meta, info):
+        """Populate attributes."""
+        self.mnemonic_identifier = mnemonic_identifier
+        self.start_time = start_time
+        self.end_time = end_time
+        self.data = data
+        self.meta = meta
+        self.info = info
+
+    def __str__(self):
+        """Return string describing the instance."""
+        return 'EdbMnemonic {} with {} records between {} and {}'.format(
+            self.mnemonic_identifier, len(self.data), self.start_time.isot,
+            self.end_time.isot)
+
+    # def interpolate(self, date, **kwargs):
```

Comment:
@Johannes-Sahlmann You could probably leave this uncommented actually, since you raise a `NotImplementedError`

Reply:
OK. Done.